crates/polars-io/src/csv/parser.rs

```diff
@@ -1,11 +1,38 @@
+use std::path::PathBuf;
+
 use memchr::memchr2_iter;
 use num_traits::Pow;
 use polars_core::prelude::*;
+use polars_utils::index::Bounded;
 
 use super::buffer::*;
 use crate::csv::read::NullValuesCompiled;
 use crate::csv::splitfields::SplitFields;
 use crate::csv::CommentPrefix;
+use crate::utils::get_reader_bytes;
+
+/// Read the number of rows without parsing columns
+/// useful for count(*) queries
+pub fn count_rows(
```

Comment:
If we first split chunks (like we do in the reader), and then use this function on the chunk_offsets, we have a parallel row-count. All logic should already be implemented.